frontend/src/components/Banner.tsx

```diff
@@ -0,0 +1,90 @@
+import { ReactNode } from "react";
+import { OutboundLink } from "react-ga";
+import { useLocalization } from "@fluent/react";
+import styles from "./Banner.module.scss";
+import warningIcon from "../../../static/images/icon-orange-info.svg";
+import { useLocalDismissal } from "../hooks/localDismissal";
+import { CloseIcon } from "./icons/close";
+import { useGaPing } from "../hooks/gaPing";
+
+export type BannerProps = {
+  children: ReactNode;
+  type?: "promo" | "warning";
+  title?: string;
+  illustration?: ReactNode;
+  cta?: {
+    target: string;
+    content: string;
+    onClick?: () => void;
+    gaPing?: Parameters<typeof useGaPing>[0];
+  };
+  dismissal?: {
+    key: string;
+    duration?: number;
+  },
+  hiddenWithAddon?: boolean;
+};
+
+export const Banner = (props: BannerProps) => {
+  const ctaRef = useGaPing(props.cta?.gaPing ?? null);
+  const dismissal = useLocalDismissal(props.dismissal?.key ?? "unused", { duration: props.dismissal?.duration });
+  const {l10n} = useLocalization();
+  const type = props.type ?? "warning";
+  const icon =
+    props.type === "warning" ? (
+      <img src={warningIcon.src} alt="" className={styles.icon} />
+    ) : null;
+  const title =
+    typeof props.title !== "undefined" ? (
+      <h2 className={styles.title}>
+        {icon}
+        {props.title}
+      </h2>
+    ) : null;
+
+  const illustration = props.illustration ? (
+    <div className={styles.illustration}>{props.illustration}</div>
+  ) : null;
+
+  const cta = props.cta ? (
+    <div className={styles.cta}>
+      <OutboundLink
+        to={props.cta.target}
+        eventLabel={props.cta.content}
```

Comment:
The attributes `to` and `eventLabel` are on `<OutboundLink>`, which comes from the (technically third-party, though still by Mozilla) `react-ga` library, so I don't think we can change that.

We could change the property names of `props.cta`. `target` is a bit weird since there's also the (different!) HTML attribute `target`, but I'm thinking `to` would be weird to use too:

```js
      <Banner
        title={l10n.getString("banner-download-install-extension-headline")}
        cta={{
          to:
            "https://addons.mozilla.org/firefox/addon/private-relay/?utm_source=fx-relay&utm_medium=banner&utm_campaign=install-addon",
          content: l10n.getString("banner-download-install-extension-cta"),
        }}
      >
        <p>{l10n.getString("banner-download-install-extension-copy")}</p>
      </Banner>
```

How about `href`, like in regular links?

The use of `content` is confusing in general: the literal text content of the CTA is passed as the event label (which I expect will differ per language?). This is [already the case in the current website](https://github.com/mozilla/fx-private-relay/blob/bacc498019ae3c02476c421f3c5ee4c7d46cdb53/static/js/analytics.js#L366). I could just add a separate `eventLabel` property that we always set to the same value as `content`?